target_chains/solana/programs/pyth_solana_receiver/src/lib.rs

```diff
@@ -0,0 +1,332 @@
+        let wh_message = WormholeMessage::try_from_bytes(vaa.payload.as_slice())
+            .map_err(|_| ReceiverError::InvalidWormholeMessage)?;
+        msg!("constructed wh_message {:?}", wh_message);
+        let root: MerkleRoot<Keccak160> = MerkleRoot::new(match wh_message.payload {
+            WormholePayload::Merkle(merkle_root) => merkle_root.root,
+        });
+
+        let mut count_updates = 0;
+
+        let price_updates_len = price_updates.len();
+        for price_update in price_updates {
+            let merkle_price_update =
+                from_slice::<byteorder::BE, MerklePriceUpdate>(price_update.as_slice())
+                    .map_err(|_| ReceiverError::DeserializeUpdateFailed)?;
+            let message_vec = Vec::from(merkle_price_update.message);
+            if !root.check(merkle_price_update.proof, &message_vec) {
+                return err!(ReceiverError::InvalidPriceUpdate);
+            }
+            let msg = from_slice::<byteorder::BE, Message>(&message_vec)
+                .map_err(|_| ReceiverError::InvalidAccumulatorMessage)?;
+
+            match msg {
+                Message::PriceFeedMessage(price_feed_message) => {
+                    count_updates += 1;
+                    msg!("price_feed_message: {:?}", price_feed_message);
+                }
+                Message::TwapMessage(twap_message) => {
+                    count_updates += 1;
+                    msg!("twap_message: {:?}", twap_message);
+                }
+                _ => return err!(ReceiverError::InvalidAccumulatorMessageType),
+            }
+        }
+        msg!("verified {} / {} updates", count_updates, price_updates_len);
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+pub struct DecodePostedVaa<'info> {
+    #[account(mut)]
+    pub payer:      Signer<'info>,
+    #[account(
+        constraint = (Chain::from(posted_vaa.emitter_chain()) == Solana || Chain::from(posted_vaa.emitter_chain()) == Pythnet) @ ReceiverError::EmitterChainNotSolanaOrPythnet,
+    )]
+    pub posted_vaa: Account<'info, AnchorVaa>,
+}
+
+impl crate::accounts::DecodePostedVaa {
+    pub fn populate(payer: &Pubkey, posted_vaa: &Pubkey) -> Self {
+        crate::accounts::DecodePostedVaa {
+            payer:      *payer,
+            posted_vaa: *posted_vaa,
+        }
+    }
+}
+
+#[derive(Accounts)]
+pub struct Update<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+}
+
+#[derive(Accounts)]
+#[instruction(vaa_hash: [u8; 32])]
+pub struct PostUpdates<'info> {
+    #[account(mut)]
+    pub payer:      Signer<'info>,
+    #[account(
+        constraint = Chain::from(posted_vaa.emitter_chain()) == Pythnet @ ReceiverError::EmitterChainNotSolanaOrPythnet,
+        seeds = [
+            SEED_PREFIX_POSTED_VAA,
+            &vaa_hash
+        ],
+        seeds::program = wormhole_anchor::program::Wormhole::id(),
+        bump
+    )]
```

Comment:
similar to above but im passing in the program as an account now. this should also probably be ready/checked against a `config` pda as well

Reply:
changing this again. Now it calls `Pubkey::find_program_address([seeds], wormhole_anchor::program::Wormhole::id())` and checks against the posted_vaa account's key in the ix but leaving a TODO that the program/owner of the posted_vaa will eventually be read from a config instead of hardcoded. I'm assuming this is in case we want to eventually write our own wormhole wrapper or need to update this